5.7/docker-entrypoint.sh

```diff
@@ -1,55 +1,97 @@
 #!/bin/bash
 set -e
 
+get_option () {
+	local section=$1
+	local option=$2
+	local default=$3
+	ret=$(my_print_defaults $section | grep '^--'${option}'=' | cut -d= -f2-)
+	[ -z $ret ] && ret=$default
+	echo $ret
+}
+
 if [ "${1:0:1}" = '-' ]; then
 	set -- mysqld "$@"
 fi
 
 if [ "$1" = 'mysqld' ]; then
-	# read DATADIR from the MySQL config
+	# Get config
 	DATADIR="$("$@" --verbose --help 2>/dev/null | awk '$1 == "datadir" { print $2; exit }')"
-	
+	SOCKET=$(get_option  mysqld socket "$datadir/mysql.sock")
+	PIDFILE=$(get_option mysqld pid-file "/var/run/mysqld/mysqld.pid")
+
 	if [ ! -d "$DATADIR/mysql" ]; then
 		if [ -z "$MYSQL_ROOT_PASSWORD" -a -z "$MYSQL_ALLOW_EMPTY_PASSWORD" ]; then
 			echo >&2 'error: database is uninitialized and MYSQL_ROOT_PASSWORD not set'
 			echo >&2 '  Did you forget to add -e MYSQL_ROOT_PASSWORD=... ?'
 			exit 1
 		fi
-		
-		echo 'Running mysql_install_db ...'
-		mysql_install_db --datadir="$DATADIR" --mysqld-file="$(which "$1")"
+		if [ ! -d "$DATADIR" ]; then
+			mkdir -p $DATADIR
+		fi
+		chown -R mysql:mysql "$DATADIR"
+
+		echo 'Running mysql_install_db'
+		mysql_install_db --user=mysql --datadir=$DATADIR --insecure --mysqld-file=/usr/sbin/mysqld
```

Comment:
This line was much improved in https://github.com/docker-library/mysql/pull/56/files#diff-17ed820d5141a5e8565a0431e5385daeR20 :+1: :smile: